File: bot/src/main/java/services/Slack.java
package services;

import com.mashape.unirest.http.HttpResponse;
import com.mashape.unirest.http.Unirest;
import com.mashape.unirest.http.exceptions.UnirestException;
import shared.EnvHandler;
import shared.Log;

public class Slack extends Service {
    private static final Log log = new Log(Slack.class);

    public Slack() {
        super(EnvHandler.SLACK_WEBHOOK_URL.getValue());
    }

    @Override
    public void createMessage(String message) {
        if (shouldUse()) {
            final String correctedMessage = correctMessage(message);

            if (correctedMessage.length() < 40000) {
                sendMessage(correctedMessage);
            } else {
                sendMessage(correctedMessage.substring(0, 40001));
                createMessage(correctedMessage.substring(40001));
            }
        }
    }

    @Override
    void sendMessage(String message) {
        try {
            final HttpResponse<String> response = Unirest.post(url)
                    .header("Content-Type", "application/json")
                    .body("{\"text\" : \"" + message + "\"}")
                    .asString();
            log.debug("Status Text: " + response.getStatusText() + " | Status: " + response.getStatus(), false);
        } catch (UnirestException e) {
            log.error(e.getLocalizedMessage(), true);
        }
    }
}

File: src/main/kotlin/messaging_services/Slack.kt
package messaging_services

import com.mashape.unirest.http.Unirest
import utils.EnvVariables

private const val MAX_MESSAGE_LENGTH = 40000

object Slack : MessagingService(MAX_MESSAGE_LENGTH) {
    override fun sendMessage(message: String) {
        EnvVariables.SlackWebhookUrl.variable?.let {
            val response = Unirest.post(it)
                .header("Content-Type", "application/json")
                .body("{\"text\" : \"```$message```\"}")
                .asString()
            println("Status Text: " + response.statusText + " | Status: " + response.status)
        }
    }
}